source/lmp/fix_dplr.cpp

```diff
@@ -312,25 +303,61 @@ void FixDPLR::init() {
 
 /* ---------------------------------------------------------------------- */
 
+void FixDPLR::setup_post_neighbor() {
+  double **x = atom->x;
+
+  vector<pair<int, int> > valid_pairs;
+  get_valid_pairs(valid_pairs, 1);
+
+  for (int ii = 0; ii < valid_pairs.size(); ++ii) {
+    int idx0 = valid_pairs[ii].first;
+    int idx1 = valid_pairs[ii].second;
+    int idx0_local = atom->map(atom->tag[idx0]);
+    int idx1_local = atom->map(atom->tag[idx1]);
+
+    for (int dd = 0; dd < 3; ++dd) {
+      x[idx1][dd] = x[idx0][dd];
+      x[idx0_local][dd] = x[idx0][dd];
+      x[idx1_local][dd] = x[idx0][dd];
+    }
+  }
+  int triclinic;
+  triclinic = domain->triclinic;
+  if (triclinic) {
+    domain->x2lamda(atom->nlocal);
+  }
+  domain->pbc();
+  domain->reset_box();
+  comm->setup();
+  neighbor->setup_bins();
+  comm->exchange();
+  comm->borders();
+  if (triclinic) {
+    domain->lamda2x(atom->nlocal + atom->nghost);
+  }
+
+  neighbor->build(1);
+}
+
+/* ---------------------------------------------------------------------- */
+
+void FixDPLR::setup_pre_exchange() {}
+
+/* ---------------------------------------------------------------------- */
+
 void FixDPLR::setup_pre_force(int vflag) { pre_force(vflag); }
 
 /* ---------------------------------------------------------------------- */
 
-void FixDPLR::setup(int vflag) {
-  // if (strstr(update->integrate_style,"verlet"))
-  post_force(vflag);
-  // else {
-  //   error->all(FLERR, "respa is not supported by this fix");
-  // }
-}
+void FixDPLR::setup(int vflag) { post_force(vflag); }
 
 /* ---------------------------------------------------------------------- */
 
 void FixDPLR::min_setup(int vflag) { setup(vflag); }
 
 /* ---------------------------------------------------------------------- */
 
-void FixDPLR::get_valid_pairs(vector<pair<int, int> > &pairs) {
+void FixDPLR::get_valid_pairs(vector<pair<int, int> > &pairs, int setup) {
```

Comment:
could you plz change to name to e.g. `is_setup` or `strict_check_valid_bond_pair`, and change the type to bool.

Reply:
Done!